stl/inc/vector

```diff
@@ -1675,14 +1650,34 @@ private:
     }
 
 public:
-    _CONSTEXPR20 void reserve(_CRT_GUARDOVERFLOW const size_type _Newcapacity) {
+    _CONSTEXPR20 void reserve(_CRT_GUARDOVERFLOW size_type _Newcapacity) {
         // increase capacity to _Newcapacity (without geometric growth), provide strong guarantee
         if (_Newcapacity > capacity()) { // something to do (reserve() never shrinks)
             if (_Newcapacity > max_size()) {
                 _Xlength();
             }
 
-            _Reallocate_exactly(_Newcapacity);
+            auto& _Al         = _Getal();
+            auto& _My_data    = _Mypair._Myval2;
+            pointer& _Myfirst = _My_data._Myfirst;
+            pointer& _Mylast  = _My_data._Mylast;
+
+            const auto _Size = static_cast<size_type>(_Mylast - _Myfirst);
+
+            const pointer _Newvec = _Allocate_at_least_helper(_Al, _Newcapacity);

```

Comment:
Suggestion: Replicating this code seems unnecessary and increases maintenance burden. I believe it would be better to keep `_Reallocate_exactly`, possibly renaming it slightly, and templating it so that `reserve()` can use `_Allocate_at_least_helper` while `shrink_to_fit()` can continue to use `allocate`.